pkg/nodediscovery/nodediscovery.go

```diff
@@ -440,15 +439,13 @@ func (n *NodeDiscovery) mutateNodeResource(nodeResource *ciliumv2.CiliumNode) er
 		APIVersion: "v1",
 		Kind:       "Node",
 		Name:       nodeTypes.GetName(),
-		UID:        k8sNode.UID,
+		UID:        slimNode.UID,
 	}}
-	providerID = k8sNode.Spec.ProviderID
+	providerID = slimNode.Spec.ProviderID
 
 	// Get the addresses from k8s node and add them as part of Cilium Node.
 	// Cilium Node should contain all addresses from k8s.
-	nodeInterface := k8s.ConvertToNode(k8sNode)
-	typesNode := nodeInterface.(*k8sTypes.Node)
-	k8sNodeParsed := k8s.ParseNode(typesNode, source.Unspec)
```

Comment:
__Note:__ This transformation was removed since https://github.com/cilium/cilium/pull/25282 updated `GetK8sNode()` to return a slim node instead of a k8s node.